core/src/commonMain/kotlin/com/xebia/functional/xef/llm/ChatWithFunctions.kt

```diff
@@ -92,43 +95,57 @@ suspend fun <A> ChatApi.prompt(
           requestedMemories,
           prompt.configuration.messagePolicy.addMessagesToConversation
         )
-        .mapNotNull { it.message.toolCalls?.firstOrNull()?.function?.arguments }
+        .mapNotNull {
+          val functionName = it.message.toolCalls?.firstOrNull()?.function?.name
```

Comment:
We also need the name to find the correct descriptor from the cases of the sealed class.